server/src/main/java/org/jboss/as/server/parsing/StandaloneXml_5.java

```diff
@@ -843,27 +832,37 @@ public boolean parseAccessControl(XMLExtendedStreamReader reader, ModelNode addr
             }
 
             final Attribute attribute = Attribute.forName(reader.getAttributeLocalName(i));
-            if (attribute == Attribute.PROVIDER) {
-                ModelNode provider = AccessAuthorizationResourceDefinition.PROVIDER.parse(value, reader);
-                ModelNode op = Util.getWriteAttributeOperation(accAuthzAddr,
-                        AccessAuthorizationResourceDefinition.PROVIDER.getName(), provider);
-
-                operationsList.add(op);
-            } else if (attribute == Attribute.USE_IDENTITY_ROLES) {
-                ModelNode useIdentityRoles = AccessAuthorizationResourceDefinition.USE_IDENTITY_ROLES.parse(value, reader);
-                ModelNode op = Util.getWriteAttributeOperation(accAuthzAddr,
-                        AccessAuthorizationResourceDefinition.USE_IDENTITY_ROLES.getName(), useIdentityRoles);
-
-                operationsList.add(op);
-            } else if (attribute == Attribute.PERMISSION_COMBINATION_POLICY) {
-                ModelNode provider = AccessAuthorizationResourceDefinition.PERMISSION_COMBINATION_POLICY.parse(value, reader);
-                ModelNode op = Util.getWriteAttributeOperation(accAuthzAddr,
-                        AccessAuthorizationResourceDefinition.PERMISSION_COMBINATION_POLICY.getName(), provider);
-
-                operationsList.add(op);
-            } else {
+            if (null == attribute) {
                 throw unexpectedAttribute(reader, i);
             }
+            switch (attribute) {
```

Comment:
An enum switch results in an inner class that has to be loaded. If we ever had time we'd probably try to get rid of all of the ones we have in the parsers.